### DISCUSSION TITLE
Running on Raspberry PI

### DISCUSSION THREAD

**Discussion by brianreinhold** [rm5248/JavaSerial#30] [Q&A]
Does this library work on the Raspberry PI 3?

I am getting a Nullpointer Exception on this line:

Files.copy( library, extractedLib.toPath() ); in SerialPort.java.
The Linux system shows a serial port "ttyUSB0" when the serial device is connected.

Project works fine on Windows

**Comment by rm5248** [comment]
It should work just fine on the Raspberry Pi.  There's probably an issue extracting the native code from the JAR.

You can turn up the logging(java.util.logging) to see what it is doing, or build your own copy of the native code and set the system property `com.rm5248.javaserial.lib.path` to tell it where to load the native code from.

**Reply by brianreinhold** [reply]
Turns out I was using version 0.12 of the jar.  I needed version 0.13. With 0.13 it works.
